tests/single_trajectory_test.py

```diff
@@ -0,0 +1,321 @@
+        batch: Any,
+        batch_idx: int,
+    ) -> None:
+        self.batches.append(batch)
+
+
+def run_training_flowbot(steps):
+    cfg = OmegaConf.create(
+        {
+            "dataset": {
+                "name": "flowbot",
+                "data_dir": "/home/yishu/datasets/partnet-mobility",
+                "dataset": "umpnet",
+                "model": "flowbot",
+                "batch_size": 1,
+                "lr": 1e-3,
+                "mask_input_channel": True,
+                "randomize_camera": True,
+                "seed": 42,
+            },
+            "training": {
+                "lr": 1e-3,
+                "batch_size": 1,
+                "epochs": 100,
+                "n_proc": 0,
+                "check_val_every_n_epoch": 1,
+                "trajectory_len": 1,
+                "mode": "delta",
+                "mask_input_channel": True,
+            },
+            "resources": {
+                "gpus": [1],
+                "num_workers": 0,
+            },
+            "seed": 42,
+        }
+    )
+
+    torch.use_deterministic_algorithms(True)
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
+    torch.set_float32_matmul_precision("medium")
+
+    # Run 2 epoch of original flowbot model
+    L.seed_everything(cfg.seed)
+    flowbot_datamodule = FlowBotDataModule(
+        root=cfg.dataset.data_dir,
+        batch_size=cfg.training.batch_size,
+        num_workers=cfg.resources.num_workers,
+        n_proc=cfg.training.n_proc,  # Add n_proc
+    )
+    flowbot_train_loader = flowbot_datamodule.train_dataloader()
+    # flowbot_val_loader = flowbot_datamodule.val_dataloader()
+    # flowbot_unseen_loader = flowbot_datamodule.unseen_dataloader()
+
+    # TODO: Training
+    L.seed_everything(cfg.seed)
+    mask_channel = 1 if cfg.training.mask_input_channel else 0
+    flowbot_network = pnp.PN2Dense(
+        in_channels=mask_channel, out_channels=3, p=pnp.PN2DenseParams()
+    )
+    initial_network = copy.deepcopy(flowbot_network)
+
+    callback = DatasetCallback()
+
+    L.seed_everything(cfg.seed)
+    flowbot_model = FlowPredictorTrainingModule(
+        flowbot_network, training_cfg=cfg.training
+    )
+    flowbot_trainer = L.Trainer(
+        accelerator="cpu",
+        # devices=cfg.resources.gpus,
+        # precision="16-mixed",
+        max_steps=steps,
+        callbacks=[callback],
+        deterministic=True,
+    )
+
+    L.seed_everything(cfg.seed)
+    flowbot_trainer.fit(
+        flowbot_model,
+        flowbot_train_loader,
+        # [flowbot_train_loader, flowbot_val_loader, flowbot_unseen_loader],
+    )
+
+    return (
+        flowbot_model,
+        callback.batches,
+        callback.losses,
+        initial_network,
+        flowbot_network,
+        callback.preds,
+        callback.pre_backwards_sds,
+    )
+
+
+# TODO: aggregate two settings to unit_test, then also needs to change the
+def test_check_single_trajectory():
```

Comment:
you should mark this with https://docs.pytest.org/en/7.1.x/how-to/skipping.html#skipping-test-functions so that it doesn't run